mir-ci/mir_ci/robot/libraries/WaylandHid.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+import os
+
+from mir_ci.virtual_pointer import Button, VirtualPointer
+from robot.api.deco import keyword, library
+from robot.api.exceptions import FatalError
+
+
+@library(scope="TEST", listener="SELF")
+class WaylandHid(VirtualPointer):
+    """
+    A Robot Framework library for interacting with virtual Wayland-based HIDs.
+
+    The client connects to the display upon entering the first keyword,
+    and disconnects when the test case ends.
+
+    A FatalError is raised if WAYLAND_DISPLAY is not defined.
+    If WAYLAND_DISPLAY is defined but empty, it defaults to 'wayland-0'.
+    """
+
+    def __init__(self) -> None:
+        display_name = os.environ.get("WAYLAND_DISPLAY")
+        if display_name is None:
+            raise FatalError("WAYLAND_DISPLAY is not defined.")
+        if not display_name:
+            display_name = "wayland-0"
```

Comment:
Oh and this looks weird, shouldn't we just `.get("WAYLAND_DISPLAY", "wayland-0")`?

Reply:
I agree. I misunderstood that we should handle the scenario in which `WAYLAND_DISPLAY` is defined but empty.